docs/configuring.rst

```diff
@@ -8,6 +8,93 @@ contact the Globus Compute Team via Slack.
    All configuration examples below must be customized for the user's
    allocation, Python environment, file system, etc.
 
+GlobusComputeEngine
+^^^^^^^^^^^^^^^^^^^
+
+``GlobusComputeEngine`` is the execution backend that ``GlobusCompute`` uses
+to execute functions. To execute functions at scale, ``GlobusCompute`` can be
+configured to use a range of ``Providers`` which allow it to connect to Batch schedulers like Slurm and PBSTorque
+to provision compute nodes dynamically in response to workload. These capabilities
+are largely borrowed from Parsl's ``HighThroughputExecutor`` and therefore all
+of ``HighThroughputExecutor``'s parameter options are supported as passthrough.
+
+Note::
+As of ``globus-compute-endpoint==2.12.0`` ``GlobusComputeEngine`` replaces the
+``HighThroughputEngine`` as the default executor.
+
+Here are ``GlobusComputeEngine`` specific features:
+
+Retries
++++++++
+
+Functions submitted to the ``GlobusComputeEngine`` can fail due to infrastructure
+failures, for eg, the worker execution the task might terminate due to it running
+out of memory, or all workers under a batch job could fail due to the batch job
+exiting as it reaches the walltime limit. ``GlobusComputeEngine`` can be configured
+to automatically retry these tasks by setting ``max_retries_on_system_failure=N``
+where N is the number of retries allowed.
+
+Example config snippet:
+
+.. code-block:: yaml
+
+    amqp_port: 443
+    display_name: Retry_2_times
+    engine:
+        type: GlobusComputeEngine
+        max_retries_on_system_failure: 2  # Default=0
+
+
```

Comment:
You might consider mentioning why we chose the default as 0.  Something like

> As retries represent extra cost, the default value is 0, and must be intentionally set by users as desired.

(Obviously please wordsmith that and say it better!)

Reply:
I've gone with this: 
The endpoint config sets default retries to 0 since functions can be computationally expensive, not idempotent, or leave side-effects that affect subsequent retries.